### ISSUE TITLE
[Improvement] Disable console output improvement

### ISSUE THREAD

**Issue by da411d** [javascript-obfuscator/javascript-obfuscator#691]
Today disableConsoleOutput looks like this:
```
const func = function () {};
const that = GLOBAL;
if (!that.console) {
  that.console = (function (func){
    const c = {};
    
    c.log = func;
    c.warn = func;
    c.debug = func;
    c.info = func;
    c.error = func;
    c.exception = func;
    c.table = func;
    c.trace = func;
    
    return c;
  })(func);
} else {
  that.console.log = func;
  that.console.warn = func;
  that.console.debug = func;
  that.console.info = func;
  that.console.error = func;
  that.console.exception = func;
  that.console.table = func;
  that.console.trace = func;
}
```

Yes, it just works, but if we type `console.log` in console, we see empty function:
<kbd>![image](https://user-images.githubusercontent.com/12462454/89105428-b2fe9d80-d429-11ea-8ed9-653d06b333f1.png)</kbd>

My improvement idea is to override `toString` method to make it look more realistic:
<kbd>![image](https://user-images.githubusercontent.com/12462454/89105374-3075de00-d429-11ea-90df-37c7be12052f.png)</kbd>
For this you need add just 3 lines of code to current template:
```
var toString = () => "function () { [native code] }";
toString.toString = toString;
func.toString = toString;
```

Now it looks more realistic, but still not perfect. We can bind original `toStinrg` method to new empty function:
<kbd>![image](https://user-images.githubusercontent.com/12462454/89105643-7fbd0e00-d42b-11ea-89b5-08ce0f338072.png)</kbd>

Now `console.log` looks like native function.

```
const cons = window.console;
for(let key in cons){
  const originalFunction = cons[key];
  if(typeof originalFunction !== "function") continue;
  const func = function () {};
  func.toString = originalFunction.toString.bind(originalFunction);
  cons[key] = func;
};
```

I'm not sure this improvement matters, or just "works - don't touch"

**UPD:** seems it don't work for firefox, just Chromium, but Chromium is used by more users

**Comment by sanex3339** [comment]
For me it works fine
![image](https://user-images.githubusercontent.com/9210534/89105953-5356c100-d42e-11ea-806f-2d19bb91a477.png)

**Comment by da411d** [comment]
<kbd>![image](https://user-images.githubusercontent.com/12462454/89106307-88184780-d431-11ea-9b1f-4dbe56cc93bc.png)</kbd>

**Comment by sanex3339** [comment]
Hi. Take a look at the PR. https://github.com/javascript-obfuscator/javascript-obfuscator/pull/692

**Comment by da411d** [comment]
There can be problems with Content-Security-Policy, because for browser `Function()` is equal to `eval`.

![image](https://user-images.githubusercontent.com/12462454/89377552-c2683a00-d6fa-11ea-8225-ac0056714950.png)

**Comment by da411d** [comment]
Also `console` has more than 20 methods, but you override only 7 of them.
I think you have a reason for this, I would like to hear your opinion)

**Comment by sanex3339** [comment]
> There can be problems with Content-Security-Policy, because for browser `Function()` is equal to `eval`.
> 
> ![image](https://user-images.githubusercontent.com/12462454/89377552-c2683a00-d6fa-11ea-8225-ac0056714950.png)

Oh, that's true

**Comment by sanex3339** [comment]
> Also `console` has more than 20 methods, but you override only 7 of them.
> I think you have a reason for this, I would like to hear your opinion)

I have to cover case when `console` object does not exist.
https://github.com/javascript-obfuscator/webpack-obfuscator/issues/13

As the result, I need to describe the array with the list of `console` methods that I want to replace.
To make this list smaller I describe only main methods of `console` object

**Comment by sanex3339** [comment]
Will use 
```
const func = {callControllerFunctionName}.constructor.prototype;
```

**Comment by da411d** [comment]
> Will use constructor.prototype

![image](https://user-images.githubusercontent.com/12462454/89385980-1f6aec80-d709-11ea-80a7-6e3fac44e5f9.png)

**Comment by sanex3339** [comment]
```
const consoleLogDisableFunctionName = callControllerFunctionName(this, function () {
        var that = this;

        const consoleObject = (that.console = that.console || {});
        const methods = ['log', 'warn', 'info', 'error', 'exception', 'table', 'trace'];

        for (let index = 0; index < methods.length; index++){
            const func = callControllerFunctionName.constructor.prototype.bind(callControllerFunctionName);
            const methodName = methods[index];
            const originalFunction = consoleObject[methodName] || func;

            func.toString = originalFunction.toString.bind(originalFunction);
            consoleObject[methodName] = func;
        }
    });

    consoleLogDisableFunctionName();
```

**Comment by sanex3339** [comment]
Hmm, need to check if this can break anything outside this template

**Comment by da411d** [comment]
That is a bit strange, but it's not critical
![image](https://user-images.githubusercontent.com/12462454/89394727-1b44cc00-d715-11ea-80f0-f772e83ce982.png)

**Comment by sanex3339** [comment]
Yeah, but I think it's not critical.

**Comment by da411d** [comment]
But it is very strange, maybe just bug in devtools

**Comment by da411d** [comment]
Why you don't just use `function(){}`?

**Comment by sanex3339** [comment]
It looks strange and can attract attention.

**Comment by sanex3339** [comment]
Released as `1.10.0`
